virtio_sound.rs: Added device configuration and common definitions

dorindabassey opened the pull request:
Added device configuration and common definitions for jack and PCM control structures from the virtio sound specifications to the header file

stefano-garzarella commented on crates/sound/src/virtio_sound.rs:
```diff
@@ -51,6 +51,106 @@ pub const VIRTIO_SND_S_IO_ERR: u32 = 0x8003;
+
+// supported PCM sample formats
+
+pub const VIRTIO_SND_PCM_FMT_IMA_ADPCM: u8 = 0;
+pub const VIRTIO_SND_PCM_FMT_MU_LAW: u8 = 1;
+pub const VIRTIO_SND_PCM_FMT_A_LAW: u8 = 2;
+pub const VIRTIO_SND_PCM_FMT_S8: u8 = 3;
+pub const VIRTIO_SND_PCM_FMT_U8: u8 = 4;
+pub const VIRTIO_SND_PCM_FMT_S16: u16 = 5;
+pub const VIRTIO_SND_PCM_FMT_U16: u16 = 6;
+pub const VIRTIO_SND_PCM_FMT_S18_3: u32 = 7;
+pub const VIRTIO_SND_PCM_FMT_U18_3: u32 = 8;
+pub const VIRTIO_SND_PCM_FMT_S20_3: u32 = 9;
+pub const VIRTIO_SND_PCM_FMT_U20_3: u32 = 10;
+pub const VIRTIO_SND_PCM_FMT_S24_3: u32 = 11;
+pub const VIRTIO_SND_PCM_FMT_U24_3: u32 = 12;
+pub const VIRTIO_SND_PCM_FMT_S20: u32 = 13;
+pub const VIRTIO_SND_PCM_FMT_U20: u32 = 14;
+pub const VIRTIO_SND_PCM_FMT_S24: u32 = 15;
+pub const VIRTIO_SND_PCM_FMT_U24: u32 = 16;
+pub const VIRTIO_SND_PCM_FMT_S32: u32 = 17;
+pub const VIRTIO_SND_PCM_FMT_U32: u32 = 18;
+pub const VIRTIO_SND_PCM_FMT_FLOAT: u32 = 19;
+pub const VIRTIO_SND_PCM_FMT_FLOAT64: u64 = 20;
+// digital formats (width / physical width)
+pub const VIRTIO_SND_PCM_FMT_DSD_U8: u8 = 21;
+pub const VIRTIO_SND_PCM_FMT_DSD_U16: u16 = 22;
+pub const VIRTIO_SND_PCM_FMT_DSD_U32: u32 = 23;
+pub const VIRTIO_SND_PCM_FMT_IEC958_SUBFRAME: u32 = 24;
```
IIUC the spec, these should be all `u8`, because they need to be used in `virtio_snd_pcm_set_params.format`:

dorindabassey replied:
Yes, that is true, Thanks. somehow i also missed to add definitions for virtio_snd_pcm_set_params, I will add that aswell.

Commit:
virtio_sound.rs: Added device configuration and common definitions

Added device configuration and common definitions from the
virtio sound specifications to the header file

Signed-off-by: Dorinda Bassey <dbassey@redhat.com>
